src/Equinox.MessageDb/MessageDbClient.fs

```diff
@@ -65,6 +65,9 @@ type internal MessageDbWriter(connectionString: string) =
             WriteMessage.prepareCommand streamName expectedVersion e
         events |> Seq.mapi toAppendCall |> Seq.iter batch.BatchCommands.Add
         try do! batch.ExecuteNonQueryAsync(ct) :> Task
+            match onSync with
```

Comment:
does `with` move up from L75?

Reply:
I don't think so? Is the concern that someone will try to write to another stream with this connection?

Reply:
the catch is only relevant to exceptions that can emanate from L67, no?

Reply:
hm, I see it gets messy to do an early exit